manim/mobject/graph.py

```diff
@@ -1135,7 +1210,13 @@ def construct(self):
                     self.play(Uncreate(G))
 
         """
-        return Graph(list(nxgraph.nodes), list(nxgraph.edges), **kwargs)
+        if type(nxgraph) == nx.classes.graph.Graph:
+            constructor = nx.Graph
+        elif type(nxgraph) == nx.classes.digraph.DiGraph:
+            constructor = nx.DiGraph
+        return Graph(
+            list(nxgraph.nodes), list(nxgraph.edges), constructor=constructor, **kwargs
+        )
```

Comment:
Perhaps a bit clunky, but this can also be rewritten using the proposed enum:
```py
graph_type = GraphType(type(nxgraph))
if graph_type == GraphType.UNDIRECTED:
    return Graph(list(nxgraph.nodes), list(nxgraph.edges), **kwargs)
eilf graph_type == GraphType.DIRECTED:
    return DiGraph(list(nxgraph.nodes), list(nxgraph.edges), **kwargs)
```